src/utils/numAbbr.js

```diff
@@ -0,0 +1,83 @@
+/**
+ * @flow
+ * @file Function to abbreviate a number as a string in a locale-sensitive manner.
+ * @author Box
+ */
+
+import IntlMessageFormat from 'intl-messageformat';
+import { languages, numbers } from 'box-locale-data';
+
+interface NumAbbrOptions {
+    length: string;
+    locale: string;
+    numbersData: object;
+}
+
+function numAbbrWithLocale(numbersData: object, num: number, locale: string, options?: NumAbbrOptions): string {
+    if (!num) return '0';
+    let { length } = options || {};
+    length = length || 'short';
+    let exponent: number = Math.floor(num).toString().length - 1;
+    if (num < 0) {
+        exponent -= 1; // take care of the negative sign
+    }
+    const formats = numbersData[length];
+    const digits: number = exponent >= formats.length ? exponent - formats.length + 3 : formats[exponent].digits;
+    const count: number = Math.round(num / 10 ** (exponent - digits + 1));
+    const template = new IntlMessageFormat(
+        formats[exponent > formats.length ? formats.length - 1 : exponent].msg,
+        locale,
+    );
+    return template.format({ count });
+}
+
+/**
```

Comment:
Should we be using https://www.npmjs.com/package/filesize?
We already use it inside src/utils/getFileSize.js
(also need to nuke/replace src/utils/size.js)

Reply:
Hmm, I guess this utility is not for file sizes, but numbers in general. nvm.